### PR TITLE
[ANDROAPP-5396] string error updated

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java [hunk 1]
```diff
@@ -413,7 +414,9 @@ public void downloadTei(String teiUid, String enrollmentUid) {
                                         view.showBreakTheGlass(teiUid, enrollmentUid);
                                         break;
                                     default:
-                                        view.displayMessage(new D2ErrorUtils(view.getContext()).getErrorMessage(t));
+                                        view.displayMessage(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java:417]: Why don't you inject D2ErrorUtils in the presenter?

File: commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt [hunk 2]
```diff
@@ -59,7 +60,8 @@ class ResourceManager(val context: Context) {
     }
 
     fun parseD2Error(throwable: Throwable) =
-        D2ErrorUtils(getWrapperContext()).getErrorMessage(throwable)
+        D2ErrorUtils(getWrapperContext(), NetworkUtils(getWrapperContext()))
```
Comment (andresmr) [inline commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt:63]: please, inject D2ErrorUtils into the ResourceManager

**Review by andresmr — CHANGES_REQUESTED** [review]

**Review by mmmateos — APPROVED** [review]

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java:417]: Done! I've injected the resource manager as D2ErrorUtils requires the wrapperContext to provide the correct translation.

File: commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt (hunk 2 above)
Comment (Balcan) [reply commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt:63]: Can't be done. Current implementation of resource manager depends on the activity context to provide the correct string translations. Changing it is out of the scope of the this issue.

Comment (andresmr) [reply commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt:63]: OK, thanks for the clarification

**Review by andresmr — APPROVED** [review]

**Pull request merged**